tests/integration/integration-test.py

```diff
@@ -62,6 +64,191 @@ def test_get_notes_path(self, endpoint='/notes'):
+                self.check_context_type_sort(response.json()['data'])
+
+        # invalid sortKey tests returns 400
+        invalid_sources = [" ", "lastmodified", "random", "contexttype"]
+        invalid_source = random.choice(invalid_sources)
+        self.invalid_request_test(student_id, 'sortKey', invalid_source)
+
+    def check_time_sort(self, response):
+        format = '%Y-%m-%dT%H:%M:%S.%f%z'
+        if len(response) > 0:
+            last_modified = response[0]['attributes']['lastModified']
+            greater_element = datetime.strptime(last_modified, format)
+            for resource in response:
+                current_element = resource['attributes']['lastModified']
+                current_element = datetime.strptime(current_element, format)
+                self.assertGreaterEqual(greater_element, current_element)
+                greater_element = current_element
+        else:
+            self.assertEqual(response, [])
+
+    def check_alphabetical_sort(self, response, sort_key):
+        if len(response) > 0:
+            greater_element = response[0]['attributes'][sort_key]
+            for resource in response:
+                current_element = resource['attributes'][sort_key]
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        else:
+            self.assertEqual(response, [])
+
+    def check_context_type_sort(self, response):
+        if len(response) > 0:
+            attributes = response[0]['attributes']
+            greater_element = attributes['context']['contextType']
+            for resource in response:
+                attributes = resource['attributes']
+                current_element = attributes['context']['contextType']
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        else:
+            self.assertEqual(response, [])
+
+    def query_source(self, student_id, sources, endpoint='/notes'):
+        for source in sources:
+            params = {'studentId': student_id, 'sources': source}
+            response = self.get_response(params)
+            # Validating the source requested is the same source
+            for resource in response.json()['data']:
+                actual_source = resource['attributes']['source']
+                self.assertEqual(actual_source, source)
+        # Check more than one source at a time
+        if len(sources) > 0:
+            sample_numbers = random.randint(1, len(sources))
+            rand_sources = random.sample(sources, sample_numbers)
+            sources = ",".join(rand_sources)
+            params = {'studentId': student_id, 'sources': sources}
+            response = self.get_response(params)
+            for resource in response.json()['data']:
+                    actual_source = resource['attributes']['source']
+                    self.assertIn(actual_source, sources)
```

Comment:
I think you can be stricter here and assert that `actual_source` is in `random_sources`